### DISCUSSION TITLE
Reordering albums

### DISCUSSION THREAD

**Discussion by LouisLouiseKay** [funmusicplace/mirlo#242] [Ideas]
It would be great if you could : 

> Reorder albums by drag and dropping their cards in the edit view
> Select a "featured" album (with a little star logo or whatever) that would then be displayed in its own row (that would just be a new section component that would appear before "ArtistAlbums") as the first thing people see on artist page.

**Comment by simonv3** [comment]
> Reorder albums by drag and dropping their cards in the edit view 

This doesn't really make sense to me. I feel like the order should just be by release date. That's what I would expect as a user. And then to have a featured album, that makes sense.

**Reply by fluffy-critter** [reply]
I have a pair of albums, "Transitions" and "Deadnames." They have the same release date, but one is the companion album to the other. I would really like to be able to choose the order in which they appear without changing the release date.

I would also like to be able to feature specific things at the top of my page, or put less-important releases lower down. Not everything has equal precedence in my catalog.

**Comment by LouisLouiseKay** [comment]
This would be true for a blog, but this isn't publications made "on the spot in the browser and then published", there a a variety of reasons why people would want to change order : 
- put main albums before smaller releases like single or EPs (makes me think we could add categories). 
- they have gained back control of/access to past work and they want to add it to their discography but at the right chronological place.
- they're just uploading their entire catalogue at once and mistakenly uploaded one album before the other.

And that's just the basic ones but I'm sure there are other more niche reasons for having control over this.

Personally as a user I don't care about if its from latest to oldest, I will just always assume I see first what the artist considers the most important to see first at that moment. Typically in case of gaining back control of an old album after a contract with a label expires, I could see (and have seen) an artist temporarily put the old album on top for a few weeks, and once that's done, putting it back at the bottom in chronological order.

These are really common scenarios and we shouldn't assume that "the moment when an album page is created" is = to "the moment of its release".

**Reply by simonv3** [reply]
> These are really common scenarios and we shouldn't assume that "the moment when an album page is created" is = to "the moment of its release".

That's not what I said. I said release date, not created date, which is what the `publishedAt` feature is for (it's exactly why we have two values)

On your points: 
> * they have gained back control of/access to past work and they want to add it to their discography but at the right chronological place.
> * they're just uploading their entire catalogue at once and mistakenly uploaded one album before the other.

Both of these are handled by sorting albums by `publishedAt` date no? 

> * put main albums before smaller releases like single or EPs (makes me think we could add categories).

Since albums already are assigned to a category (Single, LP, etc) wouldn't it make sense to just use that? 

> Typically in case of gaining back control of an old album after a contract with a label expires, I could see (and have seen) an artist temporarily put the old album on top for a few weeks, and once that's done, putting it back at the bottom in chronological order.

This seems like it's handled by a "featured" boolean. 

I feel like this is a scenario where providing the right functionality out of the box makes more sense than allow total customization.

**Reply by LouisLouiseKay** [reply]
So one thing I've done a lot for example : 

I have bigger releases, EPs or LPs, and smaller releases (Singles / Live renditions). At the time of the release of the smaller releases, I'll want them at the end because it's my current news cycle. But quickly after that, I want them elsewhere because I'd rather people see the last EP/LP rather than the last single/live.

One takeaway from this is : as an artist, I like to change the order of how people see my work depending on context because after all this is my one stop shop to showcase my stuff, and I feel having it constrained to "timeline" like it's a blog makes it feel more like just content than some intentional place where I display the work how I see fit.

**Comment by simonv3** [comment]
Done!
